README.md

```diff
@@ -71,85 +71,136 @@ Pyserini exposes Lucene Analyzers in Python with the `Analyzer` class.
+from pyserini.analysis import pyanalysis
 
 index_utils = pyutils.IndexReaderUtils('index-robust04-20191213/')
+```
 
-# Use terms() to grab an iterator over all terms in the collection.
-# Here, we only print out the first 10.
+Use `terms()` to grab an iterator over all terms in the collection, i.e., the dictionary.
+Note that these terms are _analyzed_.
+Here, we only print out the first 10:
+
+```python
 import itertools
 for term in itertools.islice(index_utils.terms(), 10):
     print(f'{term.term} (df={term.df}, cf={term.cf})')
+```
+
+How to fetch term statistics for a particular (unanalyzed) query term, "cities" in this case:
 
-# Here's a particular query term:
+```python
 term = 'cities'
 
 # Look up its document frequency (df) and collection frequency (cf).
-# Note, we use the 'raw' (i.e., unanalyzed form):
+# Note, we use the unanalyzed form:
 df, cf = index_utils.get_term_counts(term)
 print(f'term "{term}": df={df}, cf={cf}')
+```
 
-# Analyze the term (i.e., stem it):
+What if we want to fetch term statistics for an analyzed term?
+This can be accomplished by passing in an no-op `Analyzer`:
+
+```python
+term = 'cities'
+
+# Analyze the term.
 analyzed = index_utils.analyze(term)
 print(f'The analyzed form of "{term}" is "{analyzed[0]}"')
 
-# Fetch postings list and iterate over it (note method will analyze the term by default):
+# Pass in a no-op analyzer:
+analyzer = pyanalysis.get_lucene_analyzer(stemming=False, stopwords=False)
+index_utils.get_term_counts(term, analyzer=analyzer)
+df, cf = index_utils.get_term_counts(term)
+print(f'term "{term}": df={df}, cf={cf}')
+```
+
+Here's how to fetch and traverse postings:
+
+```python
+# Fetch and traverse postings for an unanalyzed term:
 postings_list = index_utils.get_postings_list(term)
 for posting in postings_list:
     print(f'docid={posting.docid}, tf={posting.tf}, pos={posting.positions}')
 
-# Alternatively, fetch postings list for a term that has already been analyzed:
+# Fetch and traverse postings for an analyzed term:
 postings_list = index_utils.get_postings_list(analyzed[0], analyze=False)
 for posting in postings_list:
     print(f'docid={posting.docid}, tf={posting.tf}, pos={posting.positions}')
+```
+
+Here's how to fetch the document vector for a document:
 
-# Fetch the document vector:
+```python
 doc_vector = index_utils.get_document_vector('FBIS4-67701')
-# Result is a dictionary where the keys are analyzed terms (i.e., the stemmed form that 
-# was actually indexed) and the values are the term frequencies.
 print(doc_vector)
+```
+
+The result is a dictionary where the keys are the analyzed terms and the values are the term frequencies.
+To compute the tf-idf representation of a document, do something like this:
+
+```python
+tf = index_utils.get_document_vector('FBIS4-67701')
+analyzer = pyanalysis.get_lucene_analyzer(stemming=False, stopwords=False)
+df = {term: (index_utils.get_term_counts(term, analyzer=analyzer))[1] for term in tf.keys()}
```

Comment:
Should probably take the first element from the `get_term_counts`, this refers to the cf right? I think I wrote it incorrectly in an issue, sorry!

fix:
```python 
df = {term: (index_utils.get_term_counts(term, analyzer=analyzer))[0] for term in tf.keys()}
```

Reply:
yup, good catch!